scripts/o3de/o3de/engine_template.py

```diff
@@ -2300,6 +2299,75 @@ def create_gem(gem_path: pathlib.Path,
     # Register the gem with the either o3de_manifest.json, engine.json or project.json based on the gem path
     return register.register(gem_path=gem_path) if not no_register else 0
 
+def create_repo(repo_path: pathlib.Path,
+                repo_uri: str,
+                repo_name: str = None,
+                origin: str = None,
+                origin_url: str = None,
+                summary: str = None,
+                additional_info: str = None,
+                force: bool = False,
+                replace: list = None
+                ) -> int:
+
+    template_name = 'RemoteRepo'
+
+    template_path = manifest.get_registered(template_name=template_name)
+
+    template_json = template_path / 'template.json'
+  
+    # read in the template.json
+    with open(template_json) as s:
+        try:
+            template_json_data = json.load(s)
+        except KeyError as e:
+            logger.error(f'Could not read template json {template_json}: {str(e)}.')
+            return 1
+
+    if not repo_path:
+        logger.error('Destination path on local machine cannot be empty.')
+        return 1
+    if not repo_uri:
+        logger.error('Destination path on online version control uri cannot be empty.')
+        return 1
+    if not repo_path and repo_uri:
+        logger.error('Please provide repository destination path on local machine and online uri.')
+        return 1
+    
+    if not force and os.path.isdir(repo_path):
+        logger.error(f'Destination path {repo_path} already exists.')
+        return 1
+    else:
+        os.makedirs(repo_path, exist_ok=force)
+
+    if not repo_name:
+        # destination name is now the last component of the destination_path
+        repo_name = repo_path.name
+
+    # any user supplied replacements
+    replacements = list()
+    while replace:
+        replace_this = replace.pop(0)
+        with_this = replace.pop(0)
+        replacements.append((replace_this, with_this))
+
+    replacements.append(("${Name}", repo_name))
+    replacements.append(("${RepoURI}", repo_uri))
+    replacements.append(("${Origin}",  origin if origin else "${Origin}"))
+    replacements.append(("${OriginURL}",  origin_url if origin_url else "${OriginURL}"))
+    replacements.append(("${Summary}", summary if summary else "${Summary}"))
+    replacements.append(("${AdditionalInfo}", additional_info if additional_info else "${AdditionalInfo}"))
+
+    # create repo.json file
+    if _execute_template_json(template_json_data,
+                              repo_path,
+                              template_path,
+                              replacements):    
+        logger.error(f'Instantiation of the remote repository has failed.')
+        return 1
+    logger.warning('Remote Repo is successfully initiated!')
+    json_data=manifest.load_o3de_manifest()
+    return register.register_remote_repo(json_data=json_data,repo_uri=repo_uri) 
```

Comment:
```suggestion
    return register.register(repo_uri=repo_uri, force_register_with_o3de_manifest=True) 
```

Reply:
Traceback (most recent call last):
  File "C:\o3de_Projects\o3de\scripts\o3de.py", line 111, in <module>
    ret = known_args.func(known_args) if hasattr(known_args, 'func') else 1
  File "C:\o3de_Projects\o3de\scripts\o3de\o3de\engine_template.py", line 2460, in _run_create_repo
    return create_repo(args.repo_path,
  File "C:\o3de_Projects\o3de\scripts\o3de\o3de\engine_template.py", line 2366, in create_repo
    return register.register(repo_uri=repo_uri, force_register_with_o3de_manifest=True)
  File "C:\o3de_Projects\o3de\scripts\o3de\o3de\register.py", line 892, in register
    result = result or register_repo(json_data, repo_uri, remove)
  File "C:\o3de_Projects\o3de\scripts\o3de\o3de\register.py", line 604, in register_repo
    cache_file = repo.download_repo_manifest(manifest_uri)
  File "C:\o3de_Projects\o3de\scripts\o3de\o3de\repo.py", line 61, in download_repo_manifest
    parsed_uri = git_provider.get_specific_file_uri(parsed_uri)
  File "C:\o3de_Projects\o3de\scripts\o3de\o3de\github_utils.py", line 46, in get_specific_file_uri
    logger.error(f'HTTP Error {e.code} opening {api_url.geturl()}')
AttributeError: 'str' object has no attribute 'geturl'  
I would receive error message if I use force_register_with_o3de_manifest=True

Reply:
There is a bug in the github_utis.py `get_specific_file_uri` method that is trying to treat a string as URI [here](https://github.com/o3de/o3de/blob/development/scripts/o3de/o3de/github_utils.py#L38-L48.)

The error has an easy fix.
The `logger.error` calls just need to treat the api_url as a regular string variable.
```python
        except urllib.error.HTTPError as e:
            logger.error(f'HTTP Error {e.code} opening {api_url}')
        except urllib.error.URLError as e:
            logger.error(f'URL Error {e.reason} opening {api_url}')
```

If you make that change, then the `register.register` call should work.

Reply:
That's right, thank you!

Reply:
Thank you! that does solve the error code, however register.register would then call the download function that then would require to be able to access the repo_uri provided, then this error would show up ->  [ERROR] o3de.utils: HTTP Error 404 opening https://github.com/VeryKaylin-player1/o3deRemoteRepo/repo.json

Reply:
let's register the `file://{repo_path}` because the `repo_uri` will not exist, and this way the user can immediately start testing.
but, add a `no_register` option to bypass registration and for the test we should avoid copying any files outside the tmp folder so let's mock the `repo.download_repo_manifest` so we don't actually perform any download or copy while testing.